src/V3Inst.cpp

```diff
@@ -555,6 +555,10 @@ class InstStatic final {
                    + (forTristate ? "t" : "") + "__" + cellp->name() + "__" + pinp->name());
             AstVar* newvarp
                 = new AstVar(pinVarp->fileline(), AstVarType::MODULETEMP, newvarname, pinVarp);
+            // Make the temporary variable public as well if this is public
+            if (pinVarp->isSigUserRWPublic()) {
+                newvarp->sigUserRWPublic(true);
+            }
```

Comment:
Perhaps this should instead be using propagateAttrFrom, and that function should add checking this along with the other isSigPublic stuff currently in combineType?